### PR TITLE
support explain

### PR DESCRIPTION
**Pull request by hanfei1991**
@zimulala @coocood @tiancaiamao @shenli @XuHuaiyu PTAL

### CODE REVIEW

File: executor/explain.go [hunk 1]
```diff
@@ -14,149 +14,57 @@
 package executor
 
 import (
-	"strings"
-
+	"encoding/json"
```
Comment (zimulala) [inline executor/explain.go]: Here needs an empty line.

File: expression/expression.go [hunk 2]
```diff
@@ -129,11 +136,11 @@ func (col *Column) DeepCopy() Expression {
 // Schema stands for the row schema get from input.
 type Schema []*Column
 
-// ToString output the contents of schema, for debug.
-func (s Schema) ToString() string {
+// String output the contents of schema, for debug.
```
Comment (zimulala) [inline expression/expression.go]: output -> outputs

File: plan/physical_plans.go [hunk 3]
```diff
@@ -145,12 +301,43 @@ func (p *Selection) Copy() PhysicalPlan {
 	return &np
 }
 
+// MarshalJSON implements json.Marshaler interface.
+func (p *Selection) MarshalJSON() ([]byte, error) {
+	child, err := json.Marshal(p.children[0].(PhysicalPlan))
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	conds, err := json.Marshal(p.Conditions)
```
Comment (zimulala) [inline plan/physical_plans.go]: Handle this error.

**Comment by coocood** [comment]
`json.MarshalIndent` can produce more readable results.

File: plan/physical_plans.go (hunk 3 above)
Comment (hanfei1991) [inline plan/physical_plans.go]: will fix

File: executor/explain_test.go [hunk 4]
```diff
@@ -15,79 +15,221 @@ package executor_test
...
+        "rightCond": null,
+        "otherCond": null,
+        "leftPlan": {
+            "type": "TableScan",
+            "db": "test",
+            "table": "t1",
+            "desc": false,
+            "keep order": false,
+            "access condition": [
+                "\u003e(test.t1.c1,1,)"
```
Comment (coocood) [inline executor/explain_test.go]: This symbol is not readable

Comment (coocood) [inline executor/explain_test.go]: The last `,` should be removed, and better add a space after `,`, like `, `

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**